lib/internal/modules/cjs/loader.js

```diff
@@ -588,15 +603,21 @@ Module._resolveFilename = function(request, parent, isMain, options) {
   }
 
   // look up the filename first, since that's the cache key.
-  var filename = Module._findPath(request, paths, isMain);
+  filename = Module._findPath(request, paths, isMain);
   if (!filename) {
     // eslint-disable-next-line no-restricted-syntax
     var err = new Error(`Cannot find module '${request}'`);
     err.code = 'MODULE_NOT_FOUND';
     throw err;
   }
+  if (parentFilename) {
+    const tempObj = cache.get(parentFilename) || {};
+    tempObj[request] = filename;
+    cache.set(parentFilename, tempObj);
+  }
   return filename;
 };
+Module._resolveFilename.cache = new Map();
```

Comment:
Also, is using `Map` instead of a plain object (e.g. `Object.create(null)`) actually faster?

Reply:
```js
if (depth === 0) stat.cache = new Map();
```
I saw that `stat.cache` inside this file used `Map` for the cache container, based on what is considered?